nixos/modules/services/desktops/dunst.nix

```diff
@@ -0,0 +1,463 @@
+        '';
+      };
+
+      browser = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+        '';
+      };
+
+      iconPosition = mkOption {
+        type = types.str;
+        default = "off";
+        description = ''
+        '';
+      };
+
+      iconFolders = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+        '';
+      };
+
+    };
+
+
+    frame = {
+      width = mkOption {
+        type = types.int;
+        default = 3;
+        description = ''
+        '';
+      };
+
+      color = mkOption {
+        type = types.str;
+        default = ''#aaaaaa'';
+        description = ''
+        '';
+      };
+
+    };
+
+    shortcuts = {
+      close = mkOption {
+        type = types.str;
+        default = "ctrl+space";
+        description = ''
+        '';
+      };
+
+      closeAll = mkOption {
+        type = types.str;
+        default = "ctrl+shift+space";
+        description = ''
+        '';
+      };
+
+      history = mkOption {
+        type = types.str;
+        default = "ctrl+Escape";
+        description = ''
+        '';
+      };
+
+      context = mkOption {
+        type = types.str;
+        default = "ctrl+shift+period";
+        description = ''
+        '';
+      };
+
+    };
+
+    urgency = {
+      low = {
+        background = mkOption {
+          type = types.str;
+          default = ''#222222'';
+          description = ''
+          '';
+        };
+
+        foreground = mkOption {
+          type = types.str;
+          default = ''#888888'';
+          description = ''
+          '';
+        };
+
+        timeout = mkOption {
+          type = types.int;
+          default = 10;
+          description = ''
+          '';
+        };
+      };
+      normal = {
+        background = mkOption {
+          type = types.str;
+          default = ''#285577'';
+          description = ''
+          '';
+        };
+
+        foreground = mkOption {
+          type = types.str;
+          default = ''#ffffff'';
+          description = ''
+          '';
+        };
+
+        timeout = mkOption {
+          type = types.int;
+          default = 10;
+          description = ''
+          '';
+        };
+      };
+      critical = {
+        background = mkOption {
+          type = types.str;
+          default = ''#900000'';
+          description = ''
+          '';
+        };
+
+        foreground = mkOption {
+          type = types.str;
+          default = ''#ffffff'';
+          description = ''
+          '';
+        };
+
+        timeout = mkOption {
+          type = types.int;
+          default = 0;
+          description = ''
+          '';
+        };
+
+      };
+    };
+
+    extraConfig = mkOption {
+      type = types.string;
+      default = "";
+      description = "Extra configuration to append to the config file.";
+    };
+
+    package = mkOption {
```

Comment:
Okay, but you don't use it for `environment.systemPackages`.
I still think this option is not necessary (same applies to redshift)